x-pack/plugins/canvas/canvas_plugin_src/functions/common/joinRows.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { ExpressionFunction } from '../../../../../../src/legacy/core_plugins/interpreter/types';
+import { Datatable } from '../types';
+import { getFunctionErrors } from '../../strings/functions';
+
+interface Arguments {
+  column: string;
+  separator: string;
+  quote: string;
+}
+
+const escapeString = (data: string, quotechar: string): string => {
+  if (quotechar === undefined || quotechar === '') {
+    return data;
+  } else {
+    const regex = new RegExp(quotechar, 'g');
+    return data.replace(/\\/g, `\\\\`).replace(regex, `\\${quotechar}`);
+  }
+};
+
+export function joinRows(): ExpressionFunction<'joinRows', Datatable, Arguments, string> {
+  const errors = getFunctionErrors().getCell;
+  return {
+    name: 'joinRows',
+    type: 'string',
+    help: 'Join values from rows in a datatable into a string',
+    context: {
+      types: ['datatable'],
+    },
+    args: {
+      column: {
+        aliases: ['_'],
+        types: ['string'],
+        required: true,
+        help: 'Column to join values from',
+      },
+      separator: {
+        aliases: ['sep'],
+        types: ['string'],
+        default: ',',
+        help: 'Separator to use between row values',
+      },
+      quote: {
+        types: ['string'],
+        default: 'DEFAULT',
+        help: 'Quote character around values',
+      },
+    },
+    fn: (context, { column, separator, quote }) => {
+      // work around having single quote as the default value not working.
+      const quotechar = quote === 'DEFAULT' ? "'" : quote;
+      return context.rows
+        .reduce((acc, row) => {
+          const value = row[column];
+          if (typeof value === 'undefined') {
```

Comment:
I don't think checking the value is what you want here. The check is safe, but I'm not sure you ever get an undefined value in a row (if you were seeing that, I'm wrong). Checking `context.columns` before iterating the rows would probably be better:

```js
const columnMatch = context.columns.find(col => col.name === column);
if (!columnMatch) {
  throw errors.columnNotFound(column);
}
```

Reply:
I agree with @w33ble on this.

Reply:
Fixed.